test/t-integration.cpp

```diff
@@ -320,3 +321,32 @@ TEST_CASE("control_vector") {
+
+    CHECK(embeddings.size() == expected.size());
+
+    for (size_t i = 0; i < embeddings.size(); i++)
+    {
+        REQUIRE(embeddings[i] == doctest::Approx(expected[i]).epsilon(0.0001));
```

Comment:
did you try this on multiple platforms?

Reply:
nope, I'll try it tomorrow on my Nvidia GPU